app/models/user.rb

```diff
@@ -32,10 +32,24 @@ def activate
     update_attributes activated: true, activated_at: Time.zone.now
   end
 
+  def create_reset_digest
+    self.reset_token = User.new_token
+    update_attribute :reset_digest, User.digest(reset_token)
+    update_attribute :reset_sent_at, Time.zone.now
```

Comment:
I think you should code [update for `reset_digest` and `reset_sent_at` ] with one method update such as `update_columns`